dashboard/app/controllers/api/v1/user_school_infos_controller.rb

```diff
@@ -10,49 +10,47 @@ def update_last_confirmation_date
     head :no_content
   end
 
-  # PATCH /api/v1/users_school_infos/<id>/update_end_date
-  def update_end_date
-    ActiveRecord::Base.transaction do
-      @user_school_info.update!(end_date: DateTime.now, last_confirmation_date: DateTime.now)
-      @user_school_info.user.update!(properties: {last_seen_school_info_interstitial: DateTime.now})
-    end
-
-    head :no_content
-  end
-
-  # PATCH /api/v1/user_school_infos/<id>/update_school_info_id
-  # A new row is added to the user school infos table when a teacher updates current school to a different school,
-  # then the school_info_id is updated in the users table.
-
-  def update_school_info_id
-    ActiveRecord::Base.transaction do
-      school = @user_school_info.school_info.school
-      unless school
-        school = School.create!(new_school_params)
-        school.school_info.create!(school_info_params)
+  # PATCH /api/v1/users_school_infos/<id>/school_info
+  def update
```

Comment:
there is a ton of logic in this update method, including at least a couple instances of repeated logic. Can this be broken up into helper methods to make it more readable?

Reply:
How's it look now? :trollface:

Reply:
:1st_place_medal: